modules/biobank/php/containerdao.class.inc

```diff
@@ -0,0 +1,534 @@
+
+        return $stati;
+    }
+
+    /**
+     * Queries all rows of the biobank_container_parent table and returns a
+     * nested array with the Parent Container ID field as the first index, the
+     * coordinate value as the second index, and the Child Container ID as the
+     * value.
+     *
+     * If there is no coordinate, the string 'Unassigned' serves as the second
+     * index.
+     *
+     * @return array $coordinates A Nested array of Child Container IDs
+     *                            indexed by Parent Container ID and
+     *                            Coordinate value or Unassigned.
+     */
+    public function getCoordinates() : array
+    {
+        $query       = "SELECT ContainerID as childContainerId,
+                         ParentContainerID as parentContainerId,
+                         Coordinate as coordinate
+                  FROM   biobank_container_parent";
+        $result      = $this->db->pselect($query, array());
+        $coordinates = array();
+        foreach ($result as $row) {
+            if (empty($row['coordinate'])) {
+                $coordinates[$row['parentContainerId']]['Unassigned'][] = $row['childContainerId'];
+            } else {
+                $coordinates[$row['parentContainerId']][$row['coordinate']] = $row['childContainerId'];
+            }
+        }
+
+        return $coordinates;
+    }
+
+    /**
+     * This function receives a Container Object, converts it into a SQL format
+     * and inserts it into all relevant tables in the database. Certain
+     * information is then cascaded to the children of the Container.
+     *
+     * @param object $container The Container Object to be saved into the
+     *                             database.
+     *
+     * @return int $containerId The ID of the Container that was saved.
+     */
+    public function saveInstance(Container $container) : int
+    {
+        // Converts Container Object into SQL format.
+        $containerArray = $this->getArrayFromInstance($container);
+        $data = $this->getSQLFromArray($containerArray);
+
+        // Inserts or Updates the biobank_container table with given data.
+        $this->db->insertOnDuplicateUpdate(
+            'biobank_container',
+            $data['biobank_container']
+        );
+
+        $containerId = $container->getId();
+        // If update:
+        if (isset($containerId)) {
+            $oldContainer         = $this->getInstanceFromId($containerId);
+            $oldParentContainerId = $oldContainer->getParentContainerId();
+            // If there is no longer an associated Parent Container ID, delete
+            // it from the entry from the biobank_container_parent table.
+            if (isset($oldParentContainerId) && !isset($parentContainerId)) {
+                $this->db->delete(
+                    'biobank_container_parent',
+                    array('ContainerID' => $containerId)
+                );
+            }
+            // If insert:
```

Comment:
If there a sentence and/or code missing here?

Reply:
No, I think i meant to place this comment within the else statement just below. It's moved now.